### ISSUE TITLE
Compilation error in `leex` template with elixir 1.11 if no space after `do`

### ISSUE THREAD

**Issue by tcoopman** [elixir-lang/elixir#10340]
This is probably not a bug, but I didn't expect a compilation error with 1.11 if it worked before in 1.10.

### Environment

* Elixir & Erlang/OTP versions (elixir --version): 

```bash
Erlang/OTP 23 [erts-11.0] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe]

Elixir 1.11.0-rc.0 (48cee5a) (compiled with Erlang/OTP 23)
```
* Operating system: linux
### Current behavior

This code:

```elixir
    <%= if  @in_progress do%>
      <%= live_component @socket, BoardGamesWeb.GameLive.PlayingComponent, players: @players, me: @me, other_players: @other_players %>
    <% else %>
      <%= live_component @socket, BoardGamesWeb.GameLive.WaitingComponent, players: @players %>
    <% end %>
```

fails to compile:

```bash
== Compilation error in file lib/board_games_web/live/game_live.ex ==
** (SyntaxError) lib/board_games_web/live/game_live.html.leex:5:2: unexpected reserved word: end
    (eex 1.11.0-rc.0) lib/eex/compiler.ex:126: EEx.Compiler.generate_buffer/4
    (eex 1.11.0-rc.0) lib/eex/compiler.ex:65: EEx.Compiler.generate_buffer/4
    (phoenix_live_view 0.14.4) expanding macro: Phoenix.LiveView.Renderer.__before_compile__/1
    lib/board_games_web/live/game_live.ex:1: BoardGamesWeb.GameLive (module)
    (elixir 1.11.0-rc.0) lib/kernel/parallel_compiler.ex:314: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/7
```

but with elixir 1.10 it compiles just fine.

### Expected behavior

The `do%` is the issue on the first line. Changing this to `do %` fixes it. This is probably a good thing that this is now fixed.

I guess this has to do with `[EEx] Make trimming behaviour via the :trim option more consistent` in the changelog.

So this issue is more of some *information* if anyone else would run into it.

**Comment by josevalim** [comment]
Hi @tcoopman, we definitely should fix it. Thank you for your report!

**Comment by dabaer** [comment]
Hey @josevalim, will this be backported to 1.10?

I had the opposite expectation of @tcoopman while debugging https://github.com/nccgroup/sobelow/issues/74 that the compiler should have complained due to the lack of space.

Additionally, would this not be considered a syntax error? Should `do%>` be considered valid?

cc @GriffinMB (Someone beat me to it!)

**Comment by josevalim** [comment]
The bug does not exist on v1.10. It should be fixed on v1.11.1.

**Comment by dabaer** [comment]
What exactly is being uncovered by nccgroup/sobelow#74? Should the compiler raise this on 1.10?
